File: algorithms/LCG.java
public class LCG {
    public static long linearCongGenerator(long r0, long a, long c, long m, int n) {
        long r = r0;
        for (int i = 0; i < n; i++) {
            r = (a * r + c) % m;
        }
        return r;
    }

    public static void main(String[] args) {
        System.out.println(linearCongGenerator(324, 1145, 177, 2148, 3));
    }
}

File: algorithms/lcg.ts
interface LCGParams {
    r0: number;
    a: number;
    c: number;
    m: number;
    n: number;
}

function linearCongGenerator(r0: number, a: number, c: number, m: number, n: number): number {
    let r: number = r0;
    for (let i: number = 0; i < n; i++) {
        r = (a * r + c) % m;
    }
    return r;
}

export { linearCongGenerator, LCGParams };
